extensions/amp-animation/0.1/css-expr-ast.js

```diff
@@ -14,10 +14,73 @@
  * limitations under the License.
  */
 
-import {assertHttpsUrl} from '../../../src/url';
+const FINAL_URL_RE = /^(data|https)\:/i;
+const DEG_TO_RAD = 2 * Math.PI / 360;
+const GRAD_TO_RAD = Math.PI / 200;
 
-const EXTRACT_URL_RE = /url\(\s*['"]?([^()'"]*)['"]?\s*\)/i;
-const DATA_URL_RE = /^data\:/i;
+
+/**
+ * An interface that assists in CSS evaluation.
+ * @interface
+ */
+export class CssContext {
+
+  /**
+   * Returns a resolved URL. The result must be an allowed URL for execution,
+   * with HTTPS restrictions.
+   * @param {string} unusedUrl
+   * @return {string}
+   */
+  resolveUrl(unusedUrl) {}
+
+  /**
+   * Returns the value of a CSS variable or `null` if not available.
+   * @param {string} unusedVarName
+   * @return {?CssNode}
+   */
+  getVar(unusedVarName) {}
+
+  /**
+   * Returns the current font size.
+   * @return {number}
+   */
+  getCurrentFontSize() {}
```

Comment:
Yes. There will be two forms: one for current target in amp-animation and one for a specifically selected target. This method address the former.